packages/core/src/browser/tree/tree-view-welcome-widget.tsx

```diff
@@ -0,0 +1,245 @@
+                    );
+                } else {
+                    const linkedTextNodes: React.ReactNode[] | undefined = [];
+
+                    for (const [nIndex, node] of linkedTextItems.entries()) {
+                        const linkedTextKey = `${lineKey}-${nIndex}`;
+
+                        if (typeof node === 'string') {
+                            linkedTextNodes.push(
+                                this.renderTextNode(node, linkedTextKey)
+                            );
+                        } else {
+                            linkedTextNodes.push(
+                                this.renderCommandLinkNode(node, linkedTextKey)
+                            );
+                        }
+                    }
+
+                    this.viewWelcomeNodes.push(
+                        <div key={`line-${lineKey}`}>
+                            {...linkedTextNodes}
+                        </div>
+                    );
+                }
+            }
+        }
+    }
+
+    protected renderButtonNode(node: ILink, lineKey: string): React.ReactNode {
+        return (
+            <div key={`line-${lineKey}`} className="theia-WelcomeViewButtonWrapper">
+                <button title={node.title}
+                    className="theia-button theia-WelcomeViewButton"
+                    disabled={!this.isEnabledClick(node.href)}
+                    onClick={e => this.openLinkOrCommand(e, node.href)}>
+                    {node.label}
+                </button>
+            </div>
+        );
+    }
+
+    protected renderTextNode(node: string, textKey: string): React.ReactNode {
+        return <span key={`text-${textKey}`}>{node}</span>;
+    }
+
+    protected renderCommandLinkNode(node: ILink, linkKey: string): React.ReactNode {
+        return (
+            <a key={`link-${linkKey}`}
+                className={this.isEnabledClick(node.href) ? 'theia-WelcomeViewCommandLink' : 'theia-WelcomeViewDisabledCommandLink'}
+                title={node.title || ''}
+                onClick={e => this.openLinkOrCommand(e, node.href)}>
+                {node.label}
+            </a>
+        );
+    }
+
+    protected isEnabledClick(href: string): boolean {
+        if (href.includes('command:')) {
+            const command = href.replace('command:', '');
+            return this.commands.isEnabled(command);
+        }
+        return true;
+    }
+
+    protected openLinkOrCommand(event: React.MouseEvent, href: string): void {
+        event.stopPropagation();
+
+        if (href.includes('command:')) {
+            const command = href.replace('command:', '');
+            this.commands.executeCommand(command);
+        } else {
+            window.open(href, '_blank');
+        }
+    }
+
+    protected parseLinkedText(text: string): LinkedTextItem[] {
+        const result: LinkedTextItem[] = [];
+
+        const linkRegex = /\[([^\]]+)\]\(((?:https?:\/\/|command:)[^\)\s]+)(?: ("|')([^\3]+)(\3))?\)/gi;
```

Comment:
It is very important to open a CQ if you have copied code from an open-source project.

https://github.com/microsoft/vscode/blob/573e5145ae3b50523925a6f6315d373e649d1b06/src/vs/base/common/linkedText.ts#L26